### ISSUE TITLE
Error when using HF weights to finetune on NT Benchmark Tasks

### ISSUE THREAD

**Issue by ritvikgupta199** [kuleshov-group/caduceus#67]
Hi,

I am following the instructions from the previously raised [issue](https://github.com/kuleshov-group/caduceus/issues/14) and using the [hf_finetune](https://github.com/kuleshov-group/caduceus/tree/hf_finetune) branch for using HF weights to finetune the model to NT Downstream tasks. However, I'm getting a shape mismatch error when running the scripts.

I have attached my logs for my runs here.
[nt_enhancers_caduceus_ph_LR-1e-3_BATCH_SIZE-128_RC_AUG-false_3930907.log](https://github.com/user-attachments/files/18482483/nt_enhancers_caduceus_ph_LR-1e-3_BATCH_SIZE-128_RC_AUG-false_3930907.log)

Thank you for the great work!

**Comment by yair-schiff** [comment]
TL;DR, it seems you are using the Caduceus-Ph model and so you should use the following parameters in the [wrapper script](https://github.com/kuleshov-group/caduceus/blob/26143aa7992f939eae8d32c0695b7d362d69b5d2/slurm_scripts/wrapper_run_nucleotide_transformer_hf.sh#L8-L9):
```
CONJOIN_TRAIN_DECODER="false"
CONJOIN_TEST="true"
RC_AUGS="true"
```

See below for a more in depth explanation.

---

You are using the `post-hoc` (`ph`) version of the model. This model is not inherently RC equivariant, but rather at downstream task inference time, it is made to be RC **invariant** by doing two forward passes through the base embedder model and decoder head where one forward pass corresponds to the original sequence and the other pass to the RC of the sequence. The prediction head outputs for these passes are then averaged.

When using the post-hoc version, during downstream task training/fine-tuning, we use RC data augmentation. Setting `CONJOIN_TRAIN_DECODER=false` indicates that during training the model is only doing a single forward pass on one of the sequences (either the original or the RC-data augmented version). Setting `CONJOIN_TEST=true` indicates that at test time, the model will receive both sequences as an input tensor of shape `(batch_size, seq_len, 2)` where `2` refers to the fact that in the dataloader we have concatenated the forward and RC sequences along an extra dim: see [here](https://github.com/kuleshov-group/caduceus/blob/26143aa7992f939eae8d32c0695b7d362d69b5d2/src/dataloaders/datasets/nucleotide_transformer_dataset.py#L83-L97) and [here](https://github.com/kuleshov-group/caduceus/blob/26143aa7992f939eae8d32c0695b7d362d69b5d2/src/models/sequence/dna_embedding.py#L234). We then pass each strand through the model and then concatenate the outputs along a new dim to have an output of shape: `(batch_size, seq_len, hidden_dim, 2)` (see [here](https://github.com/kuleshov-group/caduceus/blob/26143aa7992f939eae8d32c0695b7d362d69b5d2/src/models/sequence/dna_embedding.py#L235-L238)). Finally, the decoder, splits this input along this new dim and runs the prediction head on each half and then averages the output, see [here](https://github.com/kuleshov-group/caduceus/blob/26143aa7992f939eae8d32c0695b7d362d69b5d2/src/tasks/decoders.py#L146-L150).

Admittedly this usage of config params is a bit confusing at first, sorry about that.
